blueflood-core/src/test/java/com/rackspacecloud/blueflood/types/BluefloodEnumRollupTest.java

```diff
@@ -0,0 +1,77 @@
+        BluefloodEnumRollup rollup = new BluefloodEnumRollup().withHashedEnumValue(40L, 10L).withHashedEnumValue(50L,10L).withHashedEnumValue(70L,10L);
+        Assert.assertTrue(rollup.getCount() == 3);
+
+        BluefloodEnumRollup rollup1 = new BluefloodEnumRollup().withHashedEnumValue(100L, 10L).withHashedEnumValue(200L,10L).withHashedEnumValue(300L,10L);
+        Assert.assertTrue(rollup1.getCount() == 3);
+
+        BluefloodEnumRollup er = BluefloodEnumRollup.buildRollupFromEnumRollups(asPoints(BluefloodEnumRollup.class, 0, 300, rollup, rollup1));
+        Assert.assertTrue(er.getCount() == 6);
+    }
+
+    @Test
+    public void testGetCountOfDuplicateValuesEnumRollup() throws IOException{
+        BluefloodEnumRollup rollup = new BluefloodEnumRollup().withHashedEnumValue(40L, 10L).withHashedEnumValue(40L,10L).withHashedEnumValue(40L,10L);
+        Assert.assertTrue(rollup.getCount() == 1);
+
+        BluefloodEnumRollup rollup1 = new BluefloodEnumRollup().withHashedEnumValue(40L, 10L).withHashedEnumValue(40L,10L).withHashedEnumValue(40L,10L);
+        Assert.assertTrue(rollup1.getCount() == 1);
+
+        BluefloodEnumRollup er = BluefloodEnumRollup.buildRollupFromEnumRollups(asPoints(BluefloodEnumRollup.class, 0, 300, rollup, rollup1));
+        Assert.assertTrue(er.getCount() == 1);
+    }
+
+    @Test
+    public void testMixOfUniqueAndDuplicateValuesEnumRollup() throws IOException {
+        BluefloodEnumRollup rollup = new BluefloodEnumRollup().withHashedEnumValue(40L, 10L).withHashedEnumValue(40L,10L).withHashedEnumValue(40L,10L);
+        Assert.assertTrue(rollup.getCount() == 1);
+
+        BluefloodEnumRollup rollup1 = new BluefloodEnumRollup().withHashedEnumValue(30L, 10L).withHashedEnumValue(40L,10L).withHashedEnumValue(50L,10L);
+        Assert.assertTrue(rollup1.getCount() == 3);
+
+        BluefloodEnumRollup er = BluefloodEnumRollup.buildRollupFromEnumRollups(asPoints(BluefloodEnumRollup.class, 0, 300, rollup, rollup1));
+        Assert.assertTrue(er.getCount() == 3);
+
+        List<Long> lfound = new ArrayList<Long>(er.getHashedEnumValuesWithCounts().keySet());
+        Collections.sort(lfound);
+
+        List<Long> lexpected = new ArrayList<Long>(rollup1.getHashedEnumValuesWithCounts().keySet());
+        Collections.sort(lexpected);
+
+        Assert.assertEquals(lexpected, lfound);
+        Map<Long, Long> map = er.getHashedEnumValuesWithCounts();
+        for (Long hash : map.keySet()) {
+            Long hashCount = map.get(hash);
+            if (hash == 30L) {
+                Assert.assertEquals(10L, (long)hashCount);
+            }
+            else if (hash == 40L) {
+                Assert.assertEquals(40L, (long)hashCount);
+            }
+            else if (hash == 50L) {
+                Assert.assertEquals(10L, (long)hashCount);
+            }
+        }
+    }
+
+    private static <T> Points<T> asPoints(Class<T> type, long initialTime, long timeDelta, T... values) {
```

Comment:
You've added multiple copies of this method.   Can it be put in a utility class somewhere?